src/write_mongodb.c

```diff
@@ -162,7 +171,7 @@ static int wm_initialize(wm_node_t *node) /* {{{ */
     uri_length = strlen(format_string) + strlen(node->user) +
                  strlen(node->passwd) + strlen(node->host) + 5 +
                  strlen(node->db) + 1;
-    if ((uri = malloc(sizeof(char) * uri_length)) == NULL) {
+    if ((uri = calloc(uri_length, sizeof(char))) == NULL) {
```

Comment:
here and below: sizeof(char) is always 1 as defined by the C standard. uri_length should be the 2nd argument, so `calloc(1, uri_length)`.